app_dart/lib/src/model/appengine/commit.dart

```diff
@@ -63,6 +64,8 @@ class Commit extends Model<String> {
   @StringProperty(propertyName: 'Branch')
   String branch;
 
+  RepositorySlug get slug => RepositorySlug(repository.split('/').first, repository.split('/').last);
```

Comment:
Any reason to still use repository here?

Reply:
Datastore needs a serialized form of `RepositorySlug` for the composite index. Hiding this detail was non-trivial, but the references can now be `Commit(..., repository: slug.fullName);`

Reply:
Can you please add that info as a comment?